wallet/src/service/Offers.ts

```diff
@@ -117,15 +123,36 @@ export const getOfferService = (
       Object.fromEntries(
         await Promise.all(
           Object.entries(proposalTemplate).map(async ([kw, entry]) => {
-            if (entry.amount && !(entry.pursePetname && entry.value)) {
-              const unserializedAmount = await E(boardIdMarshaller).unserialize(
-                entry.amount,
-              );
-              entry.pursePetname = brandToPurse.get(unserializedAmount.brand)
-                ?.pursePetname;
-              entry.value = String(unserializedAmount.value);
+            assert(entry.amount);
+            const unserializedAmount = await E(boardIdMarshaller).unserialize(
+              entry.amount,
+            );
+            const purse = brandToPurse.get(unserializedAmount.brand);
+
+            if (purse) {
+              const { pursePetname: brandPetname, displayInfo } = purse;
+              return [
+                kw,
+                { amount: unserializedAmount, brandPetname, displayInfo },
+              ];
             }
-            return [kw, entry];
+
+            const [brands, boardAux] = await Promise.all([
+              brandsP,
+              watcher.queryBoardAux<{
+                displayInfo: DisplayInfo;
+              }>([unserializedAmount.brand])[0],
+            ]);
+            const brandPetname = brands
+              .find(([_, brand]) => brand === unserializedAmount.brand)
+              ?.at(0);
+
+            const displayInfo = boardAux?.displayInfo;
+
+            return [
+              kw,
+              { amount: unserializedAmount, brandPetname, displayInfo },
+            ];
```

Comment:
no need to define `displayInfo`
```suggestion
            return [
              kw,
              { amount, brandPetname, displayInfo: boardAux?.displayInfo },
            ];
```